stdlib/config-api/src/main/java/org/ballerinalang/stdlib/config/GetConfig.java

```diff
@@ -92,4 +95,35 @@ private BMap buildBMap(Map<String, Object> section) {
 
         return map;
     }
+
+    public static Object get(Strand strand, String configKey, Object type) {
+
+        // TODO: Add a try-catch
+        switch (type.toString()) {
+            case "STRING":
+                return configRegistry.getAsString(configKey);
+            case "INT":
+                return configRegistry.getAsInt(configKey);
+            case "FLOAT":
+                return configRegistry.getAsFloat(configKey);
+            case "BOOLEAN":
+                return configRegistry.getAsBoolean(configKey);
+            case "MAP":
+                return buildMapValue(configRegistry.getAsMap(configKey));
+            default:
+                throw new IllegalStateException("invalid value type: " + type.toString());
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private static MapValue buildMapValue(Map<String, Object> section) {
+        MapValue map = new MapValueImpl<String, Object>();
+
+        section.forEach((key, val) -> {
+            if (val instanceof String || val instanceof Long || val instanceof Double || val instanceof Boolean) {
+                map.put(key, val);
```

Comment:
What about other ref types?

Reply:
For configs, we only consider value types + maps of value types.